### PR TITLE
Fixed #32678 -- Removed SECURE_BROWSER_XSS_FILTER setting.

### PR DESCRIPTION
**Pull request by timgraham**
https://code.djangoproject.com/ticket/32678#ticket

### CODE REVIEW

File: docs/ref/checks.txt [hunk 1]
```diff
@@ -446,7 +445,7 @@ The following checks are run if you use the :option:`check --deploy` option:
   set to ``True``, so your pages will not be served with an
   ``'X-Content-Type-Options: nosniff'`` header. You should consider enabling
   this header to prevent the browser from identifying content types incorrectly.
-* **security.W007**: Your :setting:`SECURE_BROWSER_XSS_FILTER` setting is not
+* **security.W007**: Your ``SECURE_BROWSER_XSS_FILTER`` setting is not
```
Comment (pope1ni) [inline docs/ref/checks.txt:448]: Perhaps it is worth saying that the feature was removed in Django 4.0 to follow on from the check being removed in Django 3.0?

File: docs/releases/4.0.txt [hunk 2]
```diff
@@ -357,6 +357,18 @@ subdomains by setting :setting:`CSRF_COOKIE_DOMAIN` (or
...
 
+``SecurityMiddleware`` no longer sets the ``X-XSS-Protection`` header
+---------------------------------------------------------------------
+
+The :class:`~django.middleware.security.SecurityMiddleware` no longer sets the
+the ``X-XSS-Protection`` header if the ``SECURE_BROWSER_XSS_FILTER`` setting is
+``True``. The setting is removed.
+
+Most modern browsers don't honor ``X-XSS-Protection`` HTTP header. If you want
+to set the header, use this line a custom middleware::
```
Comment (pope1ni) [inline docs/releases/4.0.txt:368]:
```suggestion
to set the header, use this line in a custom middleware::
```

Comment (pope1ni) [inline docs/releases/4.0.txt:368]: I think that we should first recommend that a `Content-Security-Policy` that disables `'unsafe-inline'` should be used as documented [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection). Then we should say how to enable this if support for ancient browsers that do not support CSP must be maintained.

In short, this looks like we're taking away a security feature without explaining why or what it should be replaced with.

**Review by pope1ni — CHANGES_REQUESTED** [review]

File: docs/ref/checks.txt (hunk 1 above)
Comment (felixxm) [reply docs/ref/checks.txt:448]: I think it's fine to keep the current wording.

File: docs/releases/4.0.txt (hunk 2 above)
Comment (felixxm) [reply docs/releases/4.0.txt]: I added a small note.

Comment (felixxm) [reply docs/releases/4.0.txt]: @timgraham Do you want to take a look?

Comment (timgraham) [reply docs/releases/4.0.txt]: Seems fine. The wording from Mozilla is "without allowing unsafe-inline scripts" (not sure if you intentionally dropped "scripts" here). Thanks.

Comment (felixxm) [reply docs/releases/4.0.txt]: `scripts` added :+1:

**Comment by felixxm** [comment]
@timgraham Thanks :1st_place_medal:

**Pull request merged**